### DISCUSSION TITLE
Help regarding patch

### DISCUSSION THREAD

**Discussion by tathastu871** [termux/termux-packages#8359] [Q&A] [answered]
Can anyone explain how to create patch for not git package archives. 
If SRCURL is git we can use git diff to create patch file.
But what about tarballs

**Comment by ghost** [comment, marked as answer]
Use GNU `diff`.
```
diff -uNr sourcedir sourcedir.mod > filename.patch
```

where `sourcedir` is directory original source code and `sourcedir.mod` is a directory with modified source code.

https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md
